packages/ember-glimmer/lib/ember-template-compiler/plugins/transform-each-in-into-each.js

```diff
@@ -0,0 +1,47 @@
+/**
+ @module ember
+ @submodule ember-glimmer
+*/
+
+/**
+  A Glimmer2 AST transformation that replaces all instances of
+
+  ```handlebars
+ {{#each-in iterableThing as |key value|}}
+  ```
+
+  with
+
+  ```handlebars
+ {{#each iterableThing as |value key|}}
+  ```
+
+  @private
+  @class TransformHasBlockSyntax
+*/
+
+export default function TransformEachInIntoEach() {
+  // set later within Glimmer2 to the syntax package
+  this.syntax = null;
+}
+
+/**
+  @private
+  @method transform
+  @param {AST} ast The AST to be transformed.
+*/
+TransformEachInIntoEach.prototype.transform = function TransformEachInIntoEach_transform(ast) {
+  let { traverse, builders: b } = this.syntax;
+
+  traverse(ast, {
+    BlockStatement(node) {
+      if (node.path.original === 'each-in') {
+        let [keyParam, valueParam] = node.program.blockParams;
+        node.program.blockParams = [valueParam, keyParam];
+        return b.block(b.path('each'), node.params, node.hash, node.program, node.inverse, node.loc);
+      }
+    }
+  });
```

Comment:
_ is dangerous, because it might shadow an putter variable (or self lookup) with that same name. If you do the wrapping trick I mentioned below, you could simply flip the value and memo in the iterator and not worry about rearranging the block params here